Syncer docs: adds a guide on running the syncer locally and updates slightly outdated doc

m1kola opened the pull request:
## Summary

* Updates instructions on how to enable a syncer on a physical cluster.
* Updates development guide on how to build a syncer image and run syncer in a kind cluster with a local registry.
* Adds a guide on how to run a syncer locally for development.

m1kola commented on docs/syncer.md:
````diff
@@ -11,99 +11,237 @@ users have to install a special component named [syncer](https://github.com/kcp-
 
-1. Wait for the kcp sync target to go ready.
+    ```bash
+    kubectl config use-context base
````
Hm. We could probably use `--to-context`. Will try that.

stevekuznetsov commented:
Isn't this going to conflict with https://github.com/kcp-dev/kcp/pull/1746 ?

stevekuznetsov commented:
We should try to make this in sync with the Makefile parts of https://github.com/kcp-dev/controller-runtime-example/pull/19

m1kola commented:
> Isn't this going to conflict with #1746 ?

It did conflict. I incorporated changes from this PR here too - updated the snippets with `kubectl kcp workspace create` output and moved deployment creation into the "Running a workload" section.


> We should try to make this in sync with the Makefile parts of [kcp-dev/controller-runtime-example#19](https://github.com/kcp-dev/controller-runtime-example/pull/19)

Could you elaborate please?  I'm not sure what I need to do with regards to this.

stevekuznetsov commented:
I wonder if there's some way to provide the simple(ish) `Makefile` (that we will be validating forever) in the doc, maybe in addition to some of the manual lists? I worry otherwise that this version of the doc (which is very detailed and awesome!) will also become stale in a month or two :|

m1kola commented:
@stevekuznetsov I think having development workflow automated is a good idea. I'm not sure if it is reasonable to automate user part of the doc: as an outsider one might want to know how to sync the clusters without having to read makefile.

I'll be back to compute on the 7th of September. Happy to catch up on this topic after that date.

> I worry otherwise that this version of the doc (which is very detailed and awesome!) will also become stale in a month or two :|

It is the nature of the docs :)

stevekuznetsov commented on docs/syncer.md:
```diff
@@ -11,99 +11,225 @@ users have to install a special component named [syncer](https://github.com/kcp-
+
+You can run the syncer in a kind cluster for development.
+
+1. Create a `kind` cluster with a local registry to simplify syncer development
```
Do you need the registry for this? I was able to build the image locally and use `kind load docker-image` to put it on the nodes, as long as you're not using a `:latest` tag it should go straight to storage & not try to pull.

stevekuznetsov commented on docs/syncer.md:
````diff
@@ -11,99 +11,225 @@ users have to install a special component named [syncer](https://github.com/kcp-
-```sh
-$ kubectl wait --for=condition=Ready synctarget/<mycluster>
-```
+1. Create an organisation and immediately enter it:
````
nit: this does not have to be an org

stevekuznetsov reviewed (commented):
Content of the docs looks really good! Just two small comments.

sttts commented:
/ok-to-test

sttts commented:
/approve

sttts commented:
/lgtm
